pennylane/ops/qubit/arithmetic_ops.py

```diff
@@ -305,3 +310,257 @@ def compute_decomposition(wires):
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.IntegerComparator.matrix`
+
+        Args:
+            value (int): The value :math:`L` that the state's decimal representation is compared against.
+            control_wires (Union[Wires, Sequence[int], or int]): wires to place controls on
+            geq (bool): If set to `True`, the comparison made will be :math:`n \geq L`. If `False`, the comparison
+                made will be :math:`n < L`.
+
+        Returns:
+           tensor_like: matrix representation
+
+        **Example**
+
+        >>> print(qml.IntegerComparator.compute_matrix(2, [0,1]))
+        [[1. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 1. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 1. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 1. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 1. 0. 0.]
+         [0. 0. 0. 0. 1. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 1.]
+         [0. 0. 0. 0. 0. 0. 1. 0.]]
+        >>> print(qml.IntegerComparator.compute_matrix(2, [0,1], geq=False))
+        [[0. 1. 0. 0. 0. 0. 0. 0.]
+         [1. 0. 0. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 1. 0. 0. 0. 0.]
+         [0. 0. 1. 0. 0. 0. 0. 0.]
+         [0. 0. 0. 0. 1. 0. 0. 0.]
+         [0. 0. 0. 0. 0. 1. 0. 0.]
+         [0. 0. 0. 0. 0. 0. 1. 0.]
+         [0. 0. 0. 0. 0. 0. 0. 1.]]
+        """
+
+        if value is None:
+            raise ValueError("The value to compare to must be specified.")
+        if control_wires is None:
+            raise ValueError("Must specify the control wires.")
+        if isinstance(value, int):
+            binary = "0" + str(len(control_wires)) + "b"
+
+            if geq:
+                if value > 2 ** len(control_wires) - 1:
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+
+                else:
+                    control_values_list = [
+                        format(n, binary) for n in range(value, 2 ** (len(control_wires)))
+                    ]
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+                    for control_values in control_values_list:
+                        mat = mat @ MultiControlledX.compute_matrix(
+                            control_wires, control_values=control_values
+                        )
+
+            else:
+                if value == 0:
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+
+                else:
+                    control_values_list = [format(n, binary) for n in range(value)]
+                    mat = np.eye(2 ** (len(control_wires) + 1))
+                    for control_values in control_values_list:
+                        mat = mat @ MultiControlledX.compute_matrix(
+                            control_wires, control_values=control_values
+                        )
+
+        else:
+            raise ValueError(f"The compared value must be an int. Got {type(value)}.")
```

Comment:
Minor: we could first have a check like 
```py
if not isinstance(value, int):
    raise ValueError(f"The compared value must be an int. Got {type(value)}.")
```
And then perform the required logic from the other branch. This just makes the code a bit more readable:
1. The validation check will proceed with the main logic;
2. We can remove one level of indentation.